engine/src/main/java/com/datatorrent/stram/StreamingContainerManager.java

```diff
@@ -784,10 +784,15 @@ public void monitorHeartbeat(boolean waitForRecovery)
             containerStopRequests.put(c.getExternalId(), c.getExternalId());
           }
         } else {
-          if (currentTms - sca.lastHeartbeatMillis > this.vars.heartbeatTimeoutMillis) {
+          if (sca.lastHeartbeatMillis != -1 && (currentTms - sca.lastHeartbeatMillis > this.vars.heartbeatTimeoutMillis)) {
             if (!isApplicationIdle()) {
```

Comment:
This is the question on `isApplicationIdle()` for @tweise and @PramodSSImmaneni, not directly related to the PR.